### PR TITLE
fix(medusa, admin-ui, medusa-react): Gift Card update fixes and admin UI cleanup

### PR DESCRIPTION
**Pull request by kasperkristensen**
PR that fixes issues related to updating Gift Cards in `medusa` and `medusa-react`. Also contains several minor fixes and consistency improvements to the admin UI.

### `@medusajs/medusa`
**What**
- Allows `ends_at` of GC updates to be null, giving admins the option to remove expiration dates on Gift Cards.

### `medusa-react`
**What**
- Fixes payload of `useAdminUpdateGiftCard`

### `@medusajs/admin-ui`
**What**
- A lot of minor fixes for UI.
  - Fix forms for custom gift cards
  - Ensures that there is padding at the bottom of all domain pages. To avoid making changes to the layout of `BodyCard`, which will affect much of Admin and can be difficult to ensure won't mess up the layout of different pages, I've added a temporary `Spacer` component and appended it to all domain pages.
  - Changes a lot of `ghost` buttons to `secondary` to improve consistency
  - Add same spacing between section on pages to improve consistency
  - Update some `Select` usages to `NextSelect`
  - Removes the usage of `BreadCrumbs` component and instead uses `BackButton` across all of Admin for improved consistency.
  - + other minor fixes

> Resolves CORE-1041

### CODE REVIEW

File: packages/admin-ui/ui/src/components/forms/gift-card/gift-card-ends-at-form/index.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { Controller } from "react-hook-form"
```
Comment (olivermrbl) [inline packages/admin-ui/ui/src/components/forms/gift-card/gift-card-ends-at-form/index.tsx:1]: todo: Adding/updating `ends_at` triggers a form submit

CleanShot 2023-04-02 at 11 14 47

File: packages/admin-ui/ui/src/domain/gift-cards/custom-giftcard.tsx [hunk 2]
```diff
@@ -1,161 +1,136 @@
...
+                <div className="gap-x-xsmall grid grid-cols-2">
+                  <GiftCardRegionForm form={nestedForm(form, "region")} />
+                  <GiftCardBalanceForm
+                    form={nestedForm(form, "balance")}
+                    currencyCode={currencySubscriber}
                   />
-                </CurrencyInput.Root>
+                </div>
+              </div>
+              <GiftCardEndsAtForm form={nestedForm(form, "ends_at")} />
```
Comment (olivermrbl) [inline packages/admin-ui/ui/src/domain/gift-cards/custom-giftcard.tsx:105]: comment: See other comment

File: packages/admin-ui/ui/src/domain/gift-cards/details/index.tsx [hunk 3]
```diff
@@ -76,94 +70,98 @@ const GiftCardDetails = () => {
...
-            <div className="flex justify-between">
-              <div className="mt-6 flex space-x-6 divide-x">
-                <div className="flex flex-col">
-                  <div className="inter-smaller-regular text-grey-50 mb-1">
-                    Original amount
+          <div className="gap-y-xsmall flex flex-col">
+            <BodyCard
+              className={"h-auto min-h-0 w-full"}
+              title={`${giftCard?.code}`}
+              subtitle={`Gift Card id: ${giftCard?.id}`}
```
Comment (olivermrbl) [inline packages/admin-ui/ui/src/domain/gift-cards/details/index.tsx:88]: todo: rm
```suggestion
```

File: packages/admin-ui/ui/src/domain/gift-cards/details/index.tsx [hunk 4]
```diff
@@ -76,94 +70,98 @@ const GiftCardDetails = () => {
...
                 </div>
-                <div className="flex flex-col pl-6">
-                  <div className="inter-smaller-regular text-grey-50 mb-1">
-                    Created
-                  </div>
-                  <div>
-                    {moment(giftCard?.created_at).format("DD MMM YYYY")}
-                  </div>
+                <div className="flex items-end">
+                  <Badge variant="default">{giftCard?.region?.name}</Badge>
```
Comment (olivermrbl) [inline packages/admin-ui/ui/src/domain/gift-cards/details/index.tsx:145]: thought: Should we move this to follow the other details format and add Region as title? Seems to be floating for its self here

**Review by olivermrbl — CHANGES_REQUESTED** [review]
Great rework of Gift Cards and a nice clean up @kasperkristensen!

File: packages/admin-ui/ui/src/components/forms/gift-card/gift-card-ends-at-form/index.tsx (hunk 1 above)
Comment (kasperkristensen) [reply packages/admin-ui/ui/src/components/forms/gift-card/gift-card-ends-at-form/index.tsx:1]: great catch the number selector component was defaulting its buttons to submit type so this was a problem in multiple forms.

File: packages/admin-ui/ui/src/domain/gift-cards/details/index.tsx (hunk 4 above)
Comment (kasperkristensen) [reply packages/admin-ui/ui/src/domain/gift-cards/details/index.tsx]: I like it will move it over to the rest

**Review by olivermrbl — APPROVED** [review]
LGTM! 💯

**Pull request merged**